src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -348,18 +348,73 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
                 if disk_version.lower() == 'latest':
                     disk_version = _get_latest_image_version(cmd.cli_ctx, location, disk_publisher, disk_offer,
                                                              disk_sku)
-                client = _compute_client_factory(cmd.cli_ctx)
-                response = client.virtual_machine_images.get(location, disk_publisher, disk_offer, disk_sku,
-                                                             disk_version)
-                image_reference = response.id
             else:  # error
                 raise CLIError('usage error: --image-reference should be ID or URN (publisher:offer:sku:version).')
+        else:
+            from msrestazure.tools import parse_resource_id
+            terms = parse_resource_id(image_reference)
+            disk_publisher, disk_offer, disk_sku, disk_version = \
+                terms['child_name_1'], terms['child_name_3'], terms['child_name_4'], terms['child_name_5']
+
+        client = _compute_client_factory(cmd.cli_ctx)
+        response = client.virtual_machine_images.get(location, disk_publisher, disk_offer, disk_sku,
+                                                     disk_version)
+
+        if hasattr(response, 'hyper_v_generation') and response.hyper_v_generation == 'V2' and not image_reference_lun:
+            # will set default value of hyper_v_generation
+            logger.warning('Starting Build 2023 event, az disk create command '
+                           'will deploy Trusted Launch VM by default. To know more about Trusted Launch, '
+                           'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+            if hasattr(response, 'features') and response.features \
+                    and 'SecurityType' in response.features \
+                    and response.features['SecurityType'] == 'TrustedLaunchSupported':
+                # will set default value of security_type
+                pass
```

Comment:
Personally, it is not recommended to add code that will not be used in the near future, otherwise it is easy to be confused

Reply:
I suggest that when the image supports Gen 2 but the user uses Gen 1, we need to add a warning log to prompt the user to pass in the `--hyper-v-generation V2` parameter to use Gen 2.
If the user uses Gen 1 image and the image supports Trusted Launch, but the user does not enable Trusted Launch, we should add another warning log to remind the user that the `--security-type TrustedLaunch` parameter should be passed in to enable Trusted Launch
@AjKundnani What do you think of it?

Reply:
> I suggest that when the image supports Gen 2 but the user uses Gen 1, we need to add a warning log to prompt the user to pass in the `--hyper-v-generation V2` parameter to use Gen 2. And if the image also supports the Trusted Launch, but the user does not enable Trusted Launch, we should add another warning log to remind the user that the `--security-type TrustedLaunch` parameter should be passed in to enable Trusted Launch @AjKundnani What do you think of it?

- The OS image will support only one of Gen i.e. it'll be either Gen1 (v1) or Gen 2 (v2) but not both. (Unlike VM Size which can support both v1 and v2)
- Above being said, if user is using Gen1 image, if we can add warning log `Consider upgrading security for your workloads using Azure Trusted Launch VMs. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch`
- If image is Gen2/V2 and supports Trusted Launch, ideally we should enable or set the trusted Launch parameters to true as per request, i.e., no warning log, CLI should set the parameters and update the request from Gen2 to Trusted Launch.

Reply:
> Above being said, if user is using Gen1 image, if we can add warning log Consider upgrading security for your workloads using Azure Trusted Launch VMs. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch

@AjKundnani I have a concern that our warning log does not seem to tell users how to upgrade security for Trusted Launch, could you provide a description for it?

Reply:
> If image is Gen2/V2 and supports Trusted Launch, ideally we should enable or set the trusted Launch parameters to true as per request, i.e., along with warning log, CLI should set the parameters and update the request from Gen2 to Trusted Launch. If that's feasible?

Yes, it is feasible. But I believe this is what we will do in the next Build Event

Reply:
> If you're referring to notification message, then yes we can be more prescriptive to end users in the notification. However, after build event 2023, if CLI can enable them by default?

Yes, we can

Reply:
> The OS image will support only one of Gen i.e. it'll be either Gen1 (v1) or Gen 2 (v2) but not both. (Unlike VM Size which can support both v1 and v2)

OK, I'd like to confirm whether the REST service will return an error when the user uses the Gen 2 image and the `--hyper-v-generation` parameter is passed in as `V1`? @yanzhudd  Maybe you can help test and verify this case

Reply:
> > Above being said, if user is using Gen1 image, if we can add warning log Consider upgrading security for your workloads using Azure Trusted Launch VMs. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch
> 
> @AjKundnani I have a concern that our warning log does not seem to tell users how to upgrade security for Trusted Launch, could you provide a description for it?

@zhoxing-ms - By enabling Trusted Launch security type for VMs, users will get security functionality including Secure Boot, vTPM, guest attestation enabled for their VMs, which'll offer additional security to workloads hosted on VMs. Users won't have to make any application changes to support Trusted Launch.